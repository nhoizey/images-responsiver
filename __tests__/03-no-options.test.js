'use strict';

const prettier = require('prettier');
const imagesResponsiver = require('../index');

function cleanHtml(html) {
	return prettier.format(html, { parser: 'html' });
}

describe('image without options', () => {
	test('simple image', () => {
		const content = `<!DOCTYPE html><html><body><p>content before <img src="test.png"> content after</p></body></html>`;
		const transformed = imagesResponsiver(content);
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('simple image in subfolder', () => {
		const content = `<!DOCTYPE html><html><body><p>content before <img src="folder/test.png"> content after</p></body></html>`;
		const transformed = imagesResponsiver(content);
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('simple image with absolute URL', () => {
		const content = `<!DOCTYPE html><html><body><p>content before <img src="https://example.com/folder/test.png"> content after</p></body></html>`;
		const transformed = imagesResponsiver(content);
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('width attribute superior to the fallback', () => {
		const content = `<!DOCTYPE html><html><body>
      <img src="test.png" width="789"></body></html>`;
		const transformed = imagesResponsiver(content);
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('width attribute inferior to the fallback', () => {
		const content = `<!DOCTYPE html><html><body>
      <img src="test.png" width="543"></body></html>`;
		const transformed = imagesResponsiver(content);
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('width attribute inferior to the minWidth', () => {
		const content = `<!DOCTYPE html><html><body>
      <img src="test.png" width="200"></body></html>`;
		const transformed = imagesResponsiver(content);
		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('2 images', () => {
		const content = `<!DOCTYPE html><html><body>
      <img src="test1.png" width="350" /><img src="test2.png" width="400" /></body></html>`;
		const transformed = imagesResponsiver(content);

		expect(cleanHtml(transformed)).toMatchSnapshot();
	});
});
