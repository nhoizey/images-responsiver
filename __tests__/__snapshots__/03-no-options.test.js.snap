// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image without options 2 images 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test1.png\\"
      sizes=\\"100vw\\"
      srcset=\\"test1-320.png 320w, test1-350.png 350w\\"
      src=\\"test1-350.png\\"
      width=\\"350\\"
    /><img
      data-pristine=\\"test2.png\\"
      sizes=\\"100vw\\"
      srcset=\\"test2-320.png 320w, test2-400.png 400w\\"
      src=\\"test2-400.png\\"
      width=\\"400\\"
    />
  </body>
</html>
"
`;

exports[`image without options simple image 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <p>
      content before
      <img
        data-pristine=\\"test.png\\"
        sizes=\\"100vw\\"
        srcset=\\"
          test-320.png   320w,
          test-560.png   560w,
          test-800.png   800w,
          test-1040.png 1040w,
          test-1280.png 1280w
        \\"
        src=\\"test-640.png\\"
      />
      content after
    </p>
  </body>
</html>
"
`;

exports[`image without options simple image in subfolder 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <p>
      content before
      <img
        data-pristine=\\"folder/test.png\\"
        sizes=\\"100vw\\"
        srcset=\\"
          folder/test-320.png   320w,
          folder/test-560.png   560w,
          folder/test-800.png   800w,
          folder/test-1040.png 1040w,
          folder/test-1280.png 1280w
        \\"
        src=\\"folder/test-640.png\\"
      />
      content after
    </p>
  </body>
</html>
"
`;

exports[`image without options simple image with absolute URL 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <p>
      content before
      <img
        data-pristine=\\"https://example.com/folder/test.png\\"
        sizes=\\"100vw\\"
        srcset=\\"
          https://example.com/folder/test-320.png   320w,
          https://example.com/folder/test-560.png   560w,
          https://example.com/folder/test-800.png   800w,
          https://example.com/folder/test-1040.png 1040w,
          https://example.com/folder/test-1280.png 1280w
        \\"
        src=\\"https://example.com/folder/test-640.png\\"
      />
      content after
    </p>
  </body>
</html>
"
`;

exports[`image without options width attribute inferior to the fallback 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test.png\\"
      sizes=\\"100vw\\"
      srcset=\\"test-320.png 320w, test-543.png 543w\\"
      src=\\"test-543.png\\"
      width=\\"543\\"
    />
  </body>
</html>
"
`;

exports[`image without options width attribute inferior to the minWidth 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test.png\\"
      sizes=\\"100vw\\"
      srcset=\\"test-200.png 200w\\"
      src=\\"test-200.png\\"
      width=\\"200\\"
    />
  </body>
</html>
"
`;

exports[`image without options width attribute superior to the fallback 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test.png\\"
      sizes=\\"100vw\\"
      srcset=\\"test-320.png 320w, test-560.png 560w, test-789.png 789w\\"
      src=\\"test-640.png\\"
      width=\\"789\\"
    />
  </body>
</html>
"
`;
